dali/python/nvidia/dali/experimental/dynamic/random.py

```diff
@@ -0,0 +1,179 @@
+        if seed is None:
+            # Use Python's random to generate a random seed
+            seed = _random.randint(0, 2**31 - 1)
+        self._rng = _random.Random(seed)
+        self._seed = seed
+
+    def __call__(self):
+        """Generate a random uint32 value.
+
+        Returns
+        -------
+        int
+            A random uint32 value (as Python int, but in range [0, 2^32-1]).
+        """
+        # Generate a random value in the uint32 range [0, 2^32-1]
+        # Return as Python int (numpy will convert it when creating the array)
+        return self._rng.randint(0, 0xFFFFFFFF)
+
+    @property
+    def seed(self):
+        """Get the seed used to initialize this RNG."""
+        return self._seed
+
+    def clone(self):
+        """Create a new RNG with the same seed.
+
+        Returns
+        -------
+        RNG
+            A new RNG instance initialized with the same seed as this one.
+            This allows creating independent RNG streams that produce the same
+            sequence of random numbers.
+
+        Examples
+        --------
+        >>> import nvidia.dali.experimental.dynamic as ndd
+        >>>
+        >>> # Create an RNG
+        >>> rng1 = ndd.random.rng(seed=1234)
+        >>>
+        >>> # Clone it to create an independent copy
+        >>> rng2 = rng1.clone()
+        >>>
+        >>> # Both will generate the same sequence
+        >>> for i in range(10):
+        >>>     assert rng1() == rng2()
+        """
+        return RNG(seed=self._seed)
+
+    def __repr__(self):
+        return f"RNG(seed={self._seed})"
+
+
+# Thread-local storage for the default RNG
+_thread_local = _threading.local()
+
+
+def get_default_rng():
+    """Get the default RNG for the current thread.
+
+    Returns
+    -------
+    RNG
+        The default RNG for the current thread.
+
+    Examples
+    --------
+    >>> import nvidia.dali.experimental.dynamic as ndd
+    >>>
+    >>> # Get the default RNG
+    >>> default = ndd.random.get_default_rng()
+    >>> print(default)
+    """
+    if not hasattr(_thread_local, "default_rng"):
+        _thread_local.default_rng = RNG()
+    return _thread_local.default_rng
+
+
+def set_seed(seed):
+    """Set the seed for the default thread-local RNG.
+
+    This affects all subsequent calls to random operators that don't specify
+    an explicit RNG.
+
+    Parameters
+    ----------
+    seed : int
+        Seed for the random number generator.
+
+    Examples
+    --------
+    >>> import nvidia.dali.experimental.dynamic as ndd
+    >>>
+    >>> # Set the seed for reproducible results
+    >>> ndd.random.set_seed(1234)
+    >>> result1 = ndd.random.uniform(range=(-1, 1), shape=[10])
+    >>>
+    >>> # Reset to the same seed
+    >>> ndd.random.set_seed(1234)
+    >>> result2 = ndd.random.uniform(range=(-1, 1), shape=[10])
+    >>> # result1 and result2 should be identical
+    """
+    _thread_local.default_rng = RNG(seed=seed)
+
+
+def rng(seed=None):
```

Comment:
Why do we need this alias for the constructor?

Reply:
You are right. Removing